.github/workflows/build.yml

```diff
@@ -94,36 +110,42 @@ jobs:
           fail-on-cache-miss: true
 
       - name: Download sentry-native (Windows)
-        if: ${{ (env.CI_PUBLISHING_BUILD == 'true') || (runner.os == 'Windows') }}
+        if: ${{ (env.CI_PUBLISHING_BUILD == 'true') || (runner.os == 'Windows' && runner.arch == 'X64') }}
         uses: actions/cache/restore@v4
         with:
           path: src/Sentry/Platforms/Native/sentry-native
           key: sentry-native-Windows-${{ hashFiles('scripts/build-sentry-native.ps1') }}-${{ hashFiles('.git/modules/modules/sentry-native/HEAD') }}
           fail-on-cache-miss: true
           enableCrossOsArchive: true
 
-      - name: Setup Environment
-        uses: ./.github/actions/environment
+      - name: Download sentry-native (Windows arm64)
+        if: ${{ (env.CI_PUBLISHING_BUILD == 'true') || (runner.os == 'Windows' && runner.arch == 'ARM64') }}
+        uses: actions/cache/restore@v4
+        with:
+          path: src/Sentry/Platforms/Native/sentry-native
+          key: sentry-native-Windows-arm64-${{ hashFiles('scripts/build-sentry-native.ps1') }}-${{ hashFiles('.git/modules/modules/sentry-native/HEAD') }}
+          fail-on-cache-miss: true
+          enableCrossOsArchive: true
 
       - name: Build Native Dependencies
         uses: ./.github/actions/buildnative
 
       - name: Restore .NET Dependencies
-        run: dotnet restore Sentry-CI-Build-${{ runner.os }}.slnf --nologo
+        run: dotnet restore Sentry-CI-Build-${{ matrix.arch && format('{0}-{1}', runner.os, matrix.arch) || runner.os }}.slnf --nologo
```

Comment:
idea: maybe we could define solution filter names from the matrix to make it easier to understand?

```yaml
matrix:
  include:
    - os: ubuntu-22.04
      slnf: Sentry-CI-Build-Linux.slnf
    - os: macos-15
      slnf: Sentry-CI-Build-macOS.slnf
    - os: windows-latest
      slnf: Sentry-CI-Build-Windows.slnf
    - os: windows-11-arm
      slnf: Sentry-CI-Build-Windows-arm64.slnf

...

    - run: dotnet restore ${{ matrix.slnf }} ...
```

Reply:
Ditto, makes it easier to merge with the musl changes:
- https://github.com/getsentry/sentry-dotnet/pull/4188